spec/support/matchers/puppet_resource_should_show.rb

```diff
@@ -4,8 +4,10 @@ def puppet_resource_should_show(property_name, value)
   it "should report the correct #{property_name} value" do
     regex = if value.nil?
               /(#{property_name})(\s*)(=>)(\s*)/
+            elsif(value.is_a?(Array))
+              /(#{property_name})(\s*)(=>)(\s*)(\[#{value.sort.map!{ |v| "'#{v}'" }.join(', ')}\])/i
             else
-              /(#{property_name})(\s*)(=>)(\s*)('#{value}'|#{value})/i
+              /(#{property_name})(\s*)(=>)(\s*)('#{Regexp.escape(value)}'|#{Regexp.escape(value)})/i
```

Comment:
This will not affect other tests at current. The issue was that for the first time in our tests we are ensuring a path property is set correctly and hence we have backslashes in the path that need to be escaped otherwise the regex will fail. This Regexp.escape function is specifically for escaping strings so they can be used in regular expressions and as such should not cause any issues.